### PR TITLE
Added vmArgs preference to provide parameters

### PR DESCRIPTION
**Pull request by NikolasKomonen**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -148,6 +148,15 @@
...
           "description": "Enable/disable ability to autoclose tags. \n\n**IMPORTANT: Turn off editor.autoClosingTags for this to work\n**",
           "scope": "window"
+        },
+        "xml.vmargs": {
+          "type": [
+            "string",
+            "null"
+          ],
+          "default": "-noverify -Xmx64M -XX:+UseG1GC -XX:+UseStringDeduplication",
+          "description": "Specifies extra VM arguments used to launch the Java Language Server. Eg. use `-noverify -Xmx1G  -XX:+UseG1GC -XX:+UseStringDeduplication` to bypass class verification, increase the heap size to 1GB and enable String deduplication with the G1 Garbage collector",
```
Comment (fbricon) [inline package.json]: s/Java/XML

**Comment by fbricon** [comment]
please make the commit message more self-describing

File: package.json [hunk 2]
```diff
@@ -148,6 +148,15 @@
           "default": true,
           "description": "Enable/disable ability to autoclose tags. \n\n**IMPORTANT: Turn off editor.autoClosingTags for this to work\n**",
           "scope": "window"
+        },
+        "xml.vmargs": {
```
Comment (fbricon) [inline package.json]: xml.server.vmargs

**Comment by fbricon** [comment]
you need to detect if vmargs changed to reload the server

File: package.json (hunk 2 above)
Comment (fbricon) [reply package.json]: still not solved

File: src/extension.ts [hunk 3]
```diff
@@ -102,9 +108,28 @@ export function activate(context: ExtensionContext) {
...
+}
 
-
-
+function verifyVMArgs() {
+  let currentVMArgs = workspace.getConfiguration("xml.server").get("vmargs");
+  if(vmArgsCache != undefined) {
+    if(vmArgsCache != currentVMArgs) {
+      window.showWarningMessage(
+        "vmArgs updated, please restart window to apply changes.",
```
Comment (fbricon) [inline src/extension.ts]: XML Language server configuration changed, please reastart VS Code.

**Comment by NikolasKomonen** [comment]
@fbricon updated

**Pull request merged**